.github/actions/check-submodules/index.js

```diff
@@ -143,8 +143,18 @@ const checkSubmodules = async function () {
             if (nonCrtRepo.test(submodule.name)) {
                 const isOnMain = await isAncestor(diff.thisCommit, 'origin/main', submodule.path);
                 if (!isOnMain) {
-                    core.setFailed(`Submodule ${submodule.name} is using a branch`);
-                    return;
+                    const awslc = /^(aws-lc)$/
+                    if (awslc.test(submodule.name)) {
+                        // for aws-lc we also use fips-2022-11-02 branch for FIPS support.
+                        const isOnFIPS = await isAncestor(diff.thisCommit, 'origin/fips-2022-11-02', submodule.path);
```

Comment:
Debatable/Food for thought: Would it be better to just check `origin/fips.*` instead of hard-coding a yearly branch?

Reply:
it's 2024 now, they still use 2022, I just assume they will ever update the date

Reply:
IIRC, the FIPS branch is supposed to be a yearly snapshot/certification. We can make this change if it becomes a problem in the future.